intermediate-javascript-calorie-counter/index.js

```diff
@@ -1,84 +1,126 @@
-document.getElementById('calorie-form').onsubmit = calculate;
-
-function calculate(event) {
-  event.preventDefault();
-  clearOutput();
-
-  const total = Array.from(document.getElementsByClassName('cal-control'))
-    .map(input => Number(input.value))
-    .reduce((accumulator, currentValue) => accumulator + currentValue, 0);
+const calorieCounter = document.getElementById('calorie-counter');
+const budgetNumberInput = document.getElementById('budget');
+const entryDropdown = document.getElementById('entry-dropdown');
+const addEntryButton = document.getElementById('add-entry');
+const clearButton = document.getElementById('clear');
+const output = document.getElementById('output');
+let error = false;
+
+function cleanInputString(str) {
+  // Remove +, -, and whitespace from string
+  /*
+    Could teach the .split() and .join()
+    methods to do this with a for loop, then refactor
+    to use regex. Might also be able to have learners
+    practice with the .reverse() method here just for fun.
```

Comment:
I agree that covering all this in one function might be a bit much, and possibly discouraging if we do a bunch of refactoring into regex.

But I was looking at some common solutions for the Palindrome Checker project, and it's pretty likely that people will try to use the `.split()`, `.reverse()`, and `.join()` methods. 

It's a bit unnatural to use those just to check for a few characters, but I thought it might be a decent way to cover those concepts before learners need it.

I'll think more about this. There may be a good way to use `.split()`, a `for` loop, and `.match()` or `.test()` to do simple matches on an array of strings, then to get more fancy with a single regular expression, the global flag, and `.replace()`.